File: geqiangbao_20201117_CommandPattern_01/geqiangbao_20201117_CommandPattern_01/day1_Command/TetrisMachine/Simple2/WrapperCommand.m
//
//  WrapperCommand.m
//  geqiangbao_20201117_CommandPattern_01
//
//  Created by xiaobao on 2020/11/17.
//

#import "WrapperCommand.h"

@interface WrapperCommand ()
@property (nonatomic,strong) NSMutableArray *commands;
@end

@implementation WrapperCommand


-(instancetype)init:(NSMutableArray *)commands{
    self = [super init];
    if (self) {
        self.commands = commands;
    }
    return self;
}

-(void)execute{
    NSEnumerator *arr = [self.commands reverseObjectEnumerator];
    for (id<TetrisMachineCommandProtocol>command in arr) {
        [command execute];
    }
}

@end

File: 20.命令模式/geqiangbao_20201204_commandPattern_swift/geqiangbao_20201204_commandPattern_swift/命令模式/3.复合命令/WrapperCommand.swift
//
//  WrapperCommand.swift
//  geqiangbao_20201204_commandPattern_swift
//
//  Created by xiaobao on 2020/12/4.
//

import UIKit

class WrapperCommand: CommandProtocol {
    var commands:Array<CommandProtocol>
    
    init(commands:Array<CommandProtocol>) {
        self.commands = commands
    }
    
    func execute() {
        for command in self.commands.reversed() {
            command.execute()
        }
    }
    
}
